.github/workflows/linux-wpt.yml

```diff
@@ -62,75 +64,66 @@ jobs:
-            --filter-intermittents filtered-test-wpt.${{ matrix.chunk_id }}.json
+            --log-raw wpt-full-logs/linux-${{ inputs.wpt-layout }}/${{ matrix.chunk_id }}.log \
+            --log-raw-unexpected wpt-filtered-logs/linux-${{ inputs.wpt-layout }}/${{ matrix.chunk_id }}.log \
+            --filter-intermittents wpt-filtered-logs/linux-${{ inputs.wpt-layout }}/${{ matrix.chunk_id }}.json
         env:
           GITHUB_CONTEXT: ${{ toJson(github) }}
-          INTERMITTENT_TRACKER_DASHBOARD_SECRET: ${{ secrets.INTERMITTENT_TRACKER_DASHBOARD_SECRET }}
-      - name: Run tests (sync)
-        if: ${{ inputs.wpt-sync-from-upstream }}
-        run: |
-          python3 ./mach test-wpt $WPT_COMMAND_LINE_ARG \
-            --${{ inputs.profile }} --processes $(nproc) --timeout-multiplier 2 \
-            --total-chunks ${{ env.max_chunk_id }} --this-chunk ${{ matrix.chunk_id }} \
-            --log-raw test-wpt.${{ matrix.chunk_id }}.log \
-            --always-succeed
-      - name: Archive filtered results
-        uses: actions/upload-artifact@v3
-        if: ${{ always() && !inputs.wpt-sync-from-upstream }}
-        with:
-          name: wpt-filtered-results-linux-${{ inputs.wpt-layout }}
-          path: |
-            filtered-test-wpt.${{ matrix.chunk_id }}.json
-            unexpected-test-wpt.${{ matrix.chunk_id }}.log
-      - name: Archive logs
-        uses: actions/upload-artifact@v3
-        if: ${{ failure() && !inputs.wpt-sync-from-upstream }}
+      - name: Archive results (filtered)
+        uses: actions/upload-artifact@v4
         with:
-          name: wpt-logs-linux-${{ inputs.wpt-layout }}
-          path: |
-            test-wpt.${{ matrix.chunk_id }}.log
-      - name: Archive logs
-        uses: actions/upload-artifact@v3
-        if: ${{ inputs.wpt-sync-from-upstream }}
+          name: wpt-filtered-logs-linux-${{ inputs.wpt-layout }}-${{ matrix.chunk_id }}
+          path: wpt-filtered-logs/*/
+      - name: Archive results (full)
+        uses: actions/upload-artifact@v4
+        if: ${{ failure() || inputs.wpt-sync-from-upstream }}
         with:
-          name: wpt-logs-linux-${{ inputs.wpt-layout }}
-          path: |
-            test-wpt.${{ matrix.chunk_id }}.log
-            wpt-jsonsummary.${{ matrix.chunk_id }}.log
+          name: wpt-full-logs-linux-${{ inputs.wpt-layout }}-${{ matrix.chunk_id }}
+          path: wpt-full-logs/*/
 
   report-test-results:
-    name: Report WPT Results
+    name: Process WPT Results
     runs-on: ubuntu-latest
-    if: ${{ always() && !cancelled() && !inputs.wpt-sync-from-upstream }}
-    needs:
-      - "linux-wpt"
+    if: ${{ always() }}
+    needs: linux-wpt
     steps:
+      - name: Merge logs (filtered)
+        uses: actions/upload-artifact/merge@v4
+        with:
+          name: wpt-filtered-logs-linux-${{ inputs.wpt-layout }}
+          pattern: wpt-filtered-logs-linux-${{ inputs.wpt-layout }}-*
+          delete-merged: true
```

Comment:
That's was the issue I was having on my experiments, but `delete-merged` solves it.